components/list/src/list_formatter.rs

```diff
@@ -64,6 +81,16 @@ macro_rules! constructor {
     };
 }
 
+fn get_data_locale_from_prefs(prefs: ListFormatterPreferences) -> DataLocale {
```

Comment:
Ideally this should be generated as well. The way to do this would be to add the marker as a macro argument and branch on `<$marker>::INFO.fallback_config.priority`

Reply:
Spun out as https://github.com/unicode-org/icu4x/issues/5764